src/main/java/com/google/api/codegen/discovery/config/TypeNameGenerator.java

```diff
@@ -146,18 +147,27 @@ public String getStringFormatExample(String format, String dateRef, String dateT
   }
 
   /**
-   * Returns an example demonstrating the given field pattern or an empty string if pattern is
-   * invalid.
+   * Returns a string placeholder demonstrating the field or an empty string if none.
+   *
+   * <p>If pattern is valid, a path template style string is returned ("foo/my-bar"). If pattern is
+   * invalid, and inPath is false, an empty string is returned. If pattern is invalid, and inPath is
+   * true, a "my-bar" style string is returned.
    *
    * <p>If not the empty string, the returned value will be enclosed within the correct
    * language-specific quotes.
    */
-  public String getFieldPatternExample(String pattern) {
-    String def = DefaultString.getNonTrivialPlaceholder(pattern);
-    if (Strings.isNullOrEmpty(def)) {
+  public String getStringFieldPlaceholder(String fieldName, String pattern, boolean inPath) {
+    String format = "my-%s";
+    String val = DefaultString.getNonTrivialPlaceholder(pattern, format);
+    if (!Strings.isNullOrEmpty(val)) {
```

Comment:
It's theoretically possible according to the discovery doc format. This function should allow for it, it checks first to see if the pattern is valid. Otherwise it only returns a placeholder for path parameters.